go.mod

```diff
@@ -23,17 +23,17 @@ require (
 	github.com/opentracing/opentracing-go v1.1.0
 	github.com/pkg/errors v0.9.1
 	github.com/pmezard/go-difflib v1.0.0
-	github.com/prometheus/client_golang v1.12.1
-	github.com/prometheus/exporter-toolkit v0.7.2-0.20220901134540-2434b08435da
+	github.com/prometheus/client_golang v1.13.0
+	github.com/prometheus/exporter-toolkit v0.8.1
 	github.com/sercand/kuberesolver v2.4.0+incompatible
 	github.com/sirupsen/logrus v1.6.0
-	github.com/stretchr/testify v1.4.0
+	github.com/stretchr/testify v1.8.0
 	github.com/uber/jaeger-client-go v2.28.0+incompatible
 	github.com/uber/jaeger-lib v2.2.0+incompatible
 	github.com/weaveworks/promrus v1.2.0
 	go.uber.org/atomic v1.5.1 // indirect
-	golang.org/x/net v0.0.0-20220225172249-27dd8689420f
-	golang.org/x/tools v0.0.0-20200825202427-b303f430e36d
-	google.golang.org/grpc v1.31.0
+	golang.org/x/net v0.0.0-20220909164309-bea034e7d591
+	golang.org/x/tools v0.1.5
+	google.golang.org/grpc v1.47.0
```

Comment:
Can it be done without this update?  This is an enormous jump.

If not, someone needs to do forwards/backwards compatibility testing to confirm that old and new clients work against old and new servers.

Reply:
Actually mimir has been running with 1.45 grpc since 14th June https://github.com/grafana/mimir/pull/2098

Reply:
We've concluded that Mimir's usage of grpc is not proof and I'm going to reuse the server test code (httpgrpc/server_test, with a few additions) to test compatibility of versions:
client using 1.31 grpc -> server 1.47
client 1.47 -> server 1.31

Reply:
What I did to test gprc message handling compatibility:

Extracted the client and server sides of the `httpgrpc/server/server.go` and httpgrpc/server/server_test.go`  files into https://github.com/krajorama/weaveworks-common-testclient and https://github.com/krajorama/weaveworks-common-testserver .

Added a test to return the OrgId extracted on the server side in the body . And another to test sending a method different then GET.

To run a test I execute the server and copied the URL it listened on, then run the client tests with:
`weaveworks-common-testclient$ SERVER_URL=direct://127.0.0.1:35635 go test -count 1 -v ./...`

The baseline was using the latest weaveworks/common at  v0.0.0-20221125125352-88dcecd133b1, where:
```
require (
        ...
	github.com/weaveworks/common v0.0.0-20221125125352-88dcecd133b1
	google.golang.org/grpc v1.31.0
)
```

Then doing a go.mod replace to get:
```
replace github.com/weaveworks/common v0.0.0-20221125125352-88dcecd133b1 => github.com/rabenhorst/common v0.0.0-20221130173339-6d74945c9349

require (
        ...
	github.com/weaveworks/common v0.0.0-20221125125352-88dcecd133b1
	google.golang.org/grpc v1.47.0
)
```

I've run the test with all combinations: client version x server version. All tests passed.